veps/sig-compute/gpu-numa.md

```diff
@@ -0,0 +1,292 @@
+virtual machine. The mapping between physical CPUs and vCPUs is stored in the
+following memory data structure: [virt-launcher CPUTune](
+https://github.com/kubevirt/kubevirt/blob/v1.4.0/pkg/virt-launcher/virtwrap/api/schema.go#L218)
+
+```xml
+<domain type="kvm" xmlns:qemu="http://libvirt.org/schemas/domain/qemu/1.0">
+    <cputune>
+        <vcpupin vcpu="0" cpuset="1"></vcpupin>
+        <vcpupin vcpu="1" cpuset="65"></vcpupin>
+...
+```
+
+- Similarly, the association between vCPUs and NUMA node in the VM is stored in
+the following memory data structure:[virt-launcher NUMA](
+https://github.com/kubevirt/kubevirt/blob/v1.4.0/pkg/virt-launcher/virtwrap/api/schema.go#L281).
+It’s worth mentioning that we need to set
+`spec.template.spec.domain.cpu.numa.guestMappingPassthrough` in the VM CR for
+the domain XML to include the following NUMA information. For the specific VM
+CR configuration, please refer to the "API Example" section.
+
+```xml
+<domain type="kvm" xmlns:qemu="http://libvirt.org/schemas/domain/qemu/1.0">
+    <cpu mode="host-model">
+        <topology sockets="1" cores="16" threads="1"></topology>
+        <numa>
+            <cell id="0" cpus="0,1,2,3,4,5,6,7" memory="32212254720" unit="b"></cell>
+            <cell id="1" cpus="40,41,42,43,44,45,46,47" memory="32212254720" unit="b"></cell>
+        </numa>
+    </cpu>
+...
+```
+
+## API Examples
+
+Based on the proposed changes above, when we create a virtual machine with the
+following configuration, the NUMA mapping functionality can be achieved.
+
+First, open the new feature gate in the KubeVirt CR:
+```yaml
+apiVersion: kubevirt.io/v1
+kind: KubeVirt
+metadata:
+  name: kubevirt
+spec:
+  configuration:
+    developerConfiguration:
+      featureGates:
+      - GPUDeviceNUMA # the feature gate to control the enabling and disabling of this functionality
+```
+
+Second, create a GPU VM with CPU/memory NUMA settings. Once
+`vm.spec.template.spec.domain.cpu.numa` and `vm.spec.template.spec.domain.devices.gpus`
+are set, GPU NUMA will be automatically set inside the VM:
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: testvm1
+spec:
+  runStrategy: Once
+  template:
+    spec:
+      domain:
+        cpu:
+          dedicatedCpuPlacement: true   # CPU NUMA settings
+          numa:                         # (KubeVirt already supported)
+            guestMappingPassthrough: {} #
+        memory:              # memory NUMA settings
+          hugepages:         # (KubeVirt already supported)
+            pageSize: "2Mi"  #
+        devices:
+          gpus:
+          - deviceName: nvidia.com/GH100_H800
+            name: gpu-1
+          - deviceName: nvidia.com/GH100_H800
+            name: gpu-2
+```
+
+## Scalability
+
+No impact.
+
+## Security
+
+No impact.
+
+## Update/Rollback Compatibility
+
+The feature gate `GPUDeviceNUMA` can control the enabling and disabling of
```

Comment:
Please ignore the feature gate in this section. Feature-gated code is disabled by default, so it obviously doesn't affect updates. Please assume that the feature gate is enabled as part of the update. Can you think of any potential issue then?